### ISSUE TITLE
Egress protocol -1 warning should not trigger for localhost

### ISSUE THREAD

**Issue by jeffb4** [stelligent/config-lint#40]
For a description of the exception needed on checking for egress rules with protocol -1, see this issue raised for cfn_nag:

https://github.com/stelligent/cfn_nag/issues/289

**Comment by kmonihen** [comment]
I don't think this applies any longer since Terraform removes the default egress rules.
Doc: https://www.terraform.io/docs/providers/aws/r/security_group.html

> NOTE on Egress rules: By default, AWS creates an ALLOW ALL egress rule when creating a new Security Group inside of a VPC. When creating a new Security Group inside a VPC, Terraform will remove this default rule, and require you specifically re-create it if you desire that rule. We feel this leads to fewer surprises in terms of controlling your egress rules. If you desire this rule to be in place, you can use this egress block:
```
egress {
  from_port   = 0
  to_port     = 0
  protocol    = "-1"
  cidr_blocks = ["0.0.0.0/0"]
}
```
Though config-lint should probably warn if you are recreating it with that method.

**Comment by kmonihen** [comment]
SG_EGRESS_ALL_PROTOCOLS and SG_WORLD_EGRESS already warn so I think we're good here.

Test:

sg_default_egress.tf
```resource "aws_security_group" "recreate_default_egress" {
  name        = "recreate_default_egress"
  description = "Recreate the default egress rule"
  vpc_id      = "12345"

  egress {
    from_port   = 0
    to_port     = 0
    protocol    = "-1"
    cidr_blocks = ["0.0.0.0/0"]
  }

  tags = {
    Name = "default_egress"
  }
}
```

```
config-lint -terraform sg_default_egress.tf 
[
  {
    "AssertionMessage": "Not expression fails",
    "Category": "resource",
    "CreatedAt": "2020-04-13T17:17:34Z",
    "Filename": "sg_default_egress.tf",
    "LineNumber": 5,
    "ResourceID": "recreate_default_egress",
    "ResourceType": "aws_security_group",
    "RuleID": "SG_EGRESS_ALL_PROTOCOLS",
    "RuleMessage": "Best practices recommend not opening all protocols and ports to egress traffic",
    "Status": "WARNING"
  },
  {
    "AssertionMessage": "Not expression fails",
    "Category": "resource",
    "CreatedAt": "2020-04-13T17:17:34Z",
    "Filename": "sg_default_egress.tf",
    "LineNumber": 5,
    "ResourceID": "recreate_default_egress",
    "ResourceType": "aws_security_group",
    "RuleID": "SG_WORLD_EGRESS",
    "RuleMessage": "Security group should not allow egress to 0.0.0.0/0 or ::/0",
    "Status": "WARNING"
  }
]
```

**Comment by jeffb4** [comment]
![image](https://user-images.githubusercontent.com/3298329/79923687-508e0e00-83f3-11ea-96d4-70b044dbcc73.png)

**Comment by jeffb4** [comment]
Probably should still not warn for 127.0.0.1
